File: frontend/src/types/invitation.ts
export interface Invitation {
  invitation_id: number
  corp_id: number
  sender_id: number
  recipient_id: number
  recipient_username: string
  corp_name: string
  status: string
  created_at: string
}

export interface InviteSendData {
  username: string
}

File: src/schemas/invitation.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class InvitationSchema(BaseModel):
    invitation_id: int
    corp_id: int
    sender_id: int
    recipient_id: int
    recipient_username: str
    corp_name: str
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InviteSendSchema(BaseModel):
    username: str
